common/src/main/java/com/wynntils/models/players/PlayerModel.java

```diff
@@ -120,18 +139,33 @@ public void onAddPlayerToTeam(PlayerTeamEvent.Added event) {
     }
 
     private void loadUser(UUID uuid, String userName) {
+        // Avoid fetching the same user multiple times
         if (fetching.contains(uuid)) return;
-        if (errorCount >= MAX_ERRORS) {
+        if (users.containsKey(uuid) || usersWithoutWynntilsAccount.contains(uuid)) return;
+
+        if (errors.size() >= MAX_ERRORS) {
             // Athena is having problems, skip this
             return;
         }
 
+        if (userFailures.getOrDefault(uuid, 0) >= MAX_USER_ERRORS) {
+            // User has had too many failures, skip this
+            return;
+        }
+
         fetching.add(uuid); // temporary, avoid extra loads
         nameMap.put(uuid, userName);
 
         ApiResponse apiResponse = Managers.Net.callApi(UrlId.API_ATHENA_USER_INFO, Map.of("uuid", uuid.toString()));
         apiResponse.handleJsonObject(
                 json -> {
+                    if (json.has("message") && json.get("message").getAsString().equals(ATHENA_USER_NOT_FOUND)) {
+                        // This user does not exist in our database, stop requesting it
+                        fetching.remove(uuid);
+                        usersWithoutWynntilsAccount.add(uuid);
```

Comment:
```suggestion
                        usersWithoutWynntilsAccount.add(uuid);
                        fetching.remove(uuid);
```

Will slightly help with a possible race condition. (Maybe it can't happen, but I like to think in code this way)